File: 2025_03/[알고리즘 고득점Kit] H-Index/H-index.py
# 내림차순으로 정렬하고, 인용수가 논문수랑 같아지는 지점 체크(논문수 1부터 증가, idx 이용)
def solution(citations):
  answer = 0
  citations.sort(reverse=True)
  for cnt, citation in enumerate(citations):
    if citation >= cnt+1:
      answer = cnt+1
  return answer


# 좋아요 많이 받은 풀이 -> 확실히 간결한데 이해도가 떨어짐
def solution(citations):
    citations.sort(reverse=True)
    answer = max(map(min, enumerate(citations, start=1)))
    return answer

File: 2025_03/프로그래머스 - 알고리즘 고득점Kit/[알고리즘 고득점Kit] H-Index/H-index.js
function solution(citations) {
  var answer = 0;

  // 내림차순 정렬
  citations.sort((a, b) => b-a);

  // 인용 수와 논문 개수가 같아지는 지점을 체크
  [...citations].forEach((v, i) => {
    if(v >= i+1) answer = i+1
  });
  
  return answer;
}